### PR TITLE
Fix pub upgrade to work with new arguments

### PR DESCRIPTION
**Pull request by jonasfj**
Fix various pub commands..

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/commands/packages.dart [hunk 1]
```diff
@@ -20,7 +20,10 @@ import '../runner/flutter_command.dart';
 class PackagesCommand extends FlutterCommand {
   PackagesCommand() {
     addSubcommand(PackagesGetCommand('get', false));
-    addSubcommand(PackagesGetCommand('upgrade', true));
+    //addSubcommand(PackagesGetCommand('upgrade', true));
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/packages.dart:23]: Please remove commented out code.

File: packages/flutter_tools/lib/src/commands/packages.dart [hunk 2]
```diff
@@ -253,3 +258,82 @@ class PackagesPassthroughCommand extends FlutterCommand {
...
+
+      await generateLocalizationsSyntheticPackage(
+        environment: environment,
+        buildSystem: globals.buildSystem,
+      );
+    }
+
+    final List<String> subArgs = rest.toList()
+      ..removeWhere((String arg) => arg == '--');
+    await pub.interactively(
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/packages.dart:328]: @jonahwilliams Can this class be tested by mocking out `pub` here? E.g. a test could make sure the argument manipulation logic above is correct.

**Review by zanderso — COMMENTED** [review]
dbc

**Pull request merged**